src/main/scala/state/Terms.scala

```diff
@@ -2295,48 +2300,70 @@ object PredicateTrigger extends PreciseCondFlyweightFactory[(String, Term, Seq[T
 
 /* Magic wands */
 
-class MagicWandSnapshot(val abstractLhs: Term, val rhsSnapshot: Term) extends Combine(abstractLhs, rhsSnapshot) {
+/**
+ * Represents a snapshot of a magic wand, which is a map from Snap to Snap.
+ *
+ * @param abstractLhs The term that represent the snapshot of the consumed left-hand side of the magic wand.
+ *                    It is used as a bound variable in a forall quantifier.
+ * @param rhsSnapshot The term that integrates the left-hand side in the snapshot that is produced after applying the magic wand.
+ * @param wandMap The map that represents the snapshot of the magic wand. It is a variable of sort Map(Snap, Snap).
+ *                In the symbolic execution this represents a function that when we apply a magic wand
+ *                it consumes the left-hand side and uses the resulting snapshot to look up which right-hand side should be produced.
+ */
+class MagicWandSnapshot(val abstractLhs: Var, val rhsSnapshot: Term, val wandMap: Term) extends Term with ConditionalFlyweight[(Var, Term, Term), MagicWandSnapshot] {
```

Comment:
One thing I'm not sure about is why a wand snapshot carries around `val abstractLhs: Var, val rhsSnapshot: Term`: aren't these just a construct of packaging a wand and shouldn't exist outside of that? Is there something else that we need these two fields for?

Reply:
Good question. I once used those fields as a compact representation for the definition of the MWSF. But the final solution does not use that anymore. Therefore, I removed those fields.